src/Angor/Client/Pages/Settings.razor

```diff
@@ -402,12 +402,28 @@
 
         networkType = _networkConfiguration.GetNetwork().Name;
 
+        _networkService.OnStatusChanged += UpdateUI;
+
         if (!networkType.ToLower().Contains("test"))
             selectedNetwork = "mainnet";
 
         return base.OnInitializedAsync();
     }
 
+    private void UpdateUI()
+    {
+        InvokeAsync(async () =>
```

Comment:
why to await when reading from storage? we do it in many places without awaiting

Reply:
I changed it to: 

  ```  private void UpdateUI()
    {
        settingsInfo = _clientStorage.GetSettingsInfo();
        StateHasChanged();
    }```

Reply:
excellent